### PR TITLE
[google_sign_in_android] Corrects typo in logs

### PR DESCRIPTION
**Pull request by Milvintsiss**
Corrects minor typo in logs printed by plugin.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/google_sign_in/google_sign_in_android/CHANGELOG.md
Comment (camsim99) [inline packages/google_sign_in/google_sign_in_android/CHANGELOG.md:3]: Can this be slightly more specific? Something like:

```suggestion
* Corrects typos in plugin error logs.
```

File: packages/google_sign_in/google_sign_in_android/pubspec.yaml
Comment (camsim99) [inline packages/google_sign_in/google_sign_in_android/pubspec.yaml:5]: A version bump is not needed here as per https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#updating-a-changelog-that-has-a-next.

**Review by camsim99 — CHANGES_REQUESTED** [review]
Thanks for the correction! I just had a few comments. This will also need a test-exemption to land and second review to land.

CC: @stuartmorgan

File: packages/google_sign_in/google_sign_in_android/pubspec.yaml
Comment (stuartmorgan) [reply packages/google_sign_in/google_sign_in_android/pubspec.yaml:5]: This does need a version bump; it changes production code. The CHANGELOG should not use `NEXT`.

**Comment by Hixie** [comment]
We _could_ test this.

More importantly, shouldn't we not be printing those logs anyway?
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console
It could be made more helpful (e.g. explaining what line of code is wrong somehow), which would make it more actionable, in which case it might be more helpful.

**Comment by Milvintsiss** [comment]
> We _could_ test this.
> 
> More importantly, shouldn't we not be printing those logs anyway? https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console It could be made more helpful (e.g. explaining what line of code is wrong somehow), which would make it more actionable, in which case it might be more helpful.

We should keep the first one because many users probably still use clientId field to provide serverClientId (it was the old behavior before 6.0.0) and this warning is fixable.
The other should be documented in the readme of google_sign_in and not be a warning. Usage of both clientId and serverClientId is the correct way for an iOS app and will not break anything for Android, it's just not needed/used for Android.

**Review by stuartmorgan — APPROVED** [review]
LGTM

We could test this with an injectable logger, but it seems like it's probably not worth the overhead of adding that testing just for a very minor dev-only logging issue.

@Hixie Are you okay with a test exemption for this version of the PR?

**Comment by Hixie** [comment]
test-exempt: based on discussions on discord, the cost of testing this outweighs the value of avoiding regressions in this code.

**Review by camsim99 — APPROVED** [review]

**Pull request merged**
